File: sdk/python/http_client/start_profile.py
"""Start profile using stdlib http.client (no requests dependency)."""
import os
import json
import http.client
from urllib.parse import urlencode

HOST = os.getenv("MLX_LAUNCHER_HOST", "launcher.mlx.yt")
PORT = int(os.getenv("MLX_LAUNCHER_PORT", "45001"))
FOLDER_ID = os.environ["MLX_FOLDER_ID"]
PROFILE_ID = os.environ["MLX_PROFILE_ID"]
AUTOMATION = os.getenv("MLX_AUTOMATION_TYPE", "puppeteer")
HEADLESS = os.getenv("MLX_HEADLESS", "false").lower() in ("1", "true", "yes")

qs = urlencode({"automation_type": AUTOMATION, "headless_mode": str(HEADLESS).lower()})
path = f"/api/v2/profile/f/{FOLDER_ID}/p/{PROFILE_ID}/start?{qs}"

conn = http.client.HTTPSConnection(HOST, PORT, timeout=120)
conn.request("GET", path, headers={"Accept": "application/json"})
res = conn.getresponse()
body = res.read().decode()
if res.status >= 400:
    raise RuntimeError(f"{res.status} {body}")
print(json.loads(body))

File: sdk/nodejs/start_profile.mjs
/**
 * Multilogin X — Start profile (Node.js fetch)
 * Node 18+ (native fetch)
 */
const host = process.env.MLX_LAUNCHER_HOST ?? "launcher.mlx.yt";
const port = process.env.MLX_LAUNCHER_PORT ?? "45001";
const folderId = process.env.MLX_FOLDER_ID;
const profileId = process.env.MLX_PROFILE_ID;
const automation = process.env.MLX_AUTOMATION_TYPE ?? "puppeteer";
const headless = process.env.MLX_HEADLESS ?? "false";

if (!folderId || !profileId) {
  throw new Error("Set MLX_FOLDER_ID and MLX_PROFILE_ID");
}

const url =
  `https://${host}:${port}/api/v2/profile/f/${folderId}/p/${profileId}/start` +
  `?automation_type=${automation}&headless_mode=${headless}`;

const res = await fetch(url, { headers: { Accept: "application/json" } });
if (!res.ok) throw new Error(`${res.status} ${await res.text()}`);
console.log(await res.json());
